### ISSUE TITLE
Payment's extension boilerplate + short tutorial

### ISSUE THREAD

**Issue by pkarw** [DivanteLtd/vue-storefront#923]
Guys, we're so close to make Your stripe extension the de-facto standard for payment extensions :)

Could You please;
1. Accept my PR with changes to the docs (regarding installation procedure) - https://github.com/develodesign/vue-storefront-stripe
2. Change the name of the plugin to "vsf-payment-stripe" - to make the convention name 
3. Change the Component name to "UpperCamelCase" - https://github.com/develodesign/vue-storefront-stripe/tree/master/lib/components
4. Add some vuex action call instead of https://github.com/develodesign/vue-storefront-stripe/blob/c3fcc524aff3257565c03caaabde669fcd232c6c/index.js#L27
5. Publish to NPM registry
6. Create the second repo like "vsf-payment-template" or "vsf-payment-boilerplate" and add some empty module template there + short tutorial inside README.md on how to create the payment module

Would be great IF you can manage this til Thursday as we're releasing the RC2

**Comment by collymore** [comment]
This has now been published with the changes requested to https://www.npmjs.com/package/vsf-payment-stripe
We'll work on a template boilerplate and some detailed instructions next.

**Comment by pkarw** [comment]
Cool!

**Comment by thomasthune** [comment]
Ok, so for me the only payment method showing up on the checkout page is the default, "cashondelivery". As far as i can see the checkout reads payment methods from a different store, than where the payment methods are added. 
Checkout reads from the cart store and the payment methods are added to the payment store.

I am a bit unsure as to where it should be placed. It is possible to change the getter in the cart store, to load the payment methods from through the rootStore, but this seems a bit wrong to me. It would look a bit like this:
```
  paymentMethods (state) {
    return rootStore.state.payment.methods
  }
```

Instead it would seem more correct that the payment store was used exclusively instead. The naming of the stores would indicate this as well.

**Comment by thomasthune** [comment]
Using payment store instead of cart, added action and getter. If this is on the track, let me know and i will create a PR. https://github.com/thomasthune/vue-storefront/commit/2f91cbe9782834f9833d5bc95e68029833a33c46

**Comment by pkarw** [comment]
This seems like good idea to unify this. Please proceed, please test the checkout extensively before pushing out the PR ok?

**Comment by pkarw** [comment]
@collymore any progress in here? :)

**Comment by kyvaith** [comment]
Please refer to my comment: https://github.com/DivanteLtd/vue-storefront/issues/792#issuecomment-380360536 and this: https://github.com/DivanteLtd/vue-storefront/pull/822#issuecomment-380362934

**Comment by MikeSheward** [comment]
@pkarw Is it the docs that are just left that needs doing for this now? We wrote some docs somewhere, possibly in the repo. We can expand upon that and create the example repo. The example payment method at the moment is cash on delivery i think? (the most simplest payment extension example).

**Comment by pkarw** [comment]
Yes. Please update and put all the docs in one place + create a boilerplate/example extension to show other guys how to create the extensiosn

**Comment by MikeSheward** [comment]
Will do!

**Comment by pkarw** [comment]
Maciej here is the task about the payments

**Comment by pkarw** [comment]
Currently we just need the point 6 to be completed:

```
Create the second repo like "vsf-payment-template" or "vsf-payment-boilerplate" and add some empty module template there + short tutorial inside README.md on how to create the payment module
```

**Comment by pkarw** [comment]
Here is a first step on the docs/boilerplate: [Payment Integration Docs](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Payment%20Gateway%20Integration.md)
